diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/AnnotationNewLineRuleWarnTest.kt

```diff
@@ -0,0 +1,241 @@
+                LintError(1,1, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SomeAnnotation not on a single line", true),
+                LintError(1,17, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SecondAnnotation not on a single line", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation fun test good`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |class A {
+                    |   val a = 5
+                    |   
+                    |  @SomeAnnotation
+                    |  @SecondAnnotation
+                    |  fun someFunc() {
+                    |       val a = 3
+                    |  }
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation fun test good 2`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |class A {
+                    |   val a = 5
+                    |   
+                    |  @SomeAnnotation fun someFunc() {
+                    |       val a = 3
+                    |  }
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation fun test bad`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |class A {
+                    |   val a = 5
+                    |   
+                    |  @SomeAnnotation @SecondAnnotation fun someFunc() {
+                    |       val a = 3
+                    |  }
+                    |}
+                """.trimMargin(),
+                LintError(4,3, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SomeAnnotation not on a single line", true),
+                LintError(4,19, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SecondAnnotation not on a single line", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation fun test bad 2`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |class A {
+                    |   val a = 5
+                    |   
+                    |  @SomeAnnotation @SecondAnnotation 
+                    |  fun someFunc() {
+                    |       val a = 3
+                    |  }
+                    |}
+                """.trimMargin(),
+                LintError(4,3, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SomeAnnotation not on a single line", true),
+                LintError(4,19, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SecondAnnotation not on a single line", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation constructor test good`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |public class Conf
+                    |@Inject
```

Comment:
Should this example be valid? I think second part of the rule states that there should be no line break when there is a single annotation. Or have you discussed this? If this requirement is not mandatory, it should be edited in code style guide too.

Reply:
It's said that one annotation should be on one line. It's not mandatory, isn't it?

Reply:
Yes, it should be not mandatory.